src/jaegertracing/DynamicLoad.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2018 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JAEGERTRACING_DYNAMIC_LOAD_H
+#define JAEGERTRACING_DYNAMIC_LOAD_H
+
+int JaegerTracingMakeTracerFactoryFct(const char* opentracing_version,
+                                      const char* opentracing_abi_version,
+                                      const void** error_category,
+                                      void* error_message,
+                                      void** tracer_factory);
```

Comment:
You don't need the function exposes. The macro OPENTRACING_DECLARE_IMPL_FACTORY declares a variable function pointer (See for example how it's done in the LS tracer: https://github.com/lightstep/lightstep-tracer-cpp/blob/master/src/dynamic_load.cpp#L6)

Reply:
Thanks for the example!